src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/GeneratorTests.Incremental.cs

```diff
@@ -0,0 +1,362 @@
+                result.ValidateIncrementalResult(IncrementalStepRunReason.Modified, IncrementalStepRunReason.Unchanged);
+            }
+
+            [Fact]
+            public async Task RunWithDiags_Then_NoOpEdit()
+            {
+                ConfigBindingGenTestDriver driver = new ConfigBindingGenTestDriver();
+
+                ConfigBindingGenRunResult result = await driver.RunGeneratorAndUpdateCompilation(BindCallSampleCodeVariant_WithUnsupportedMember);
+                result.ValidateIncrementalResult(IncrementalStepRunReason.New, IncrementalStepRunReason.New);
+
+                // We expect different spec because diag locations are different.
+                result = await driver.RunGeneratorAndUpdateCompilation(BindCallSampleCodeVariant_WithUnsupportedMember_ReorderedInvocations);
+                result.ValidateIncrementalResult(IncrementalStepRunReason.Modified, IncrementalStepRunReason.Modified);
+
+                // No diff from previous step because type model is the same.
+                result = await driver.RunGeneratorAndUpdateCompilation(BindCallSampleCodeVariant_WithUnsupportedMember_ReorderedConfigTypeMembers);
+                result.ValidateIncrementalResult(IncrementalStepRunReason.Modified, IncrementalStepRunReason.Unchanged);
+            }
+
+            [Fact]
+            public async Task RunWithDiags_Then_EditWithNoDiags()
+            {
+                ConfigBindingGenTestDriver driver = new ConfigBindingGenTestDriver();
+
+                ConfigBindingGenRunResult result = await driver.RunGeneratorAndUpdateCompilation(BindCallSampleCodeVariant_WithUnsupportedMember);
+                result.ValidateIncrementalResult(IncrementalStepRunReason.New, IncrementalStepRunReason.New);
+
+                result = await driver.RunGeneratorAndUpdateCompilation(BindCallSampleCode);
+                result.ValidateIncrementalResult(IncrementalStepRunReason.Modified, IncrementalStepRunReason.Modified);
+            }
+
+            [Fact]
+            public async Task RunWithDiags_Then_EditWithDiags()
+            {
+                ConfigBindingGenTestDriver driver = new ConfigBindingGenTestDriver();
+
+                ConfigBindingGenRunResult result = await driver.RunGeneratorAndUpdateCompilation(BindCallSampleCodeVariant_WithUnsupportedMember);
+                result.ValidateIncrementalResult(IncrementalStepRunReason.New, IncrementalStepRunReason.New);
+
+                result = await driver.RunGeneratorAndUpdateCompilation(BindCallSampleCodeVariant_WithUnsupportedMember_WithDiffMemberName);
+                result.ValidateIncrementalResult(IncrementalStepRunReason.Modified, IncrementalStepRunReason.Modified);
+            }
+        }
+
+        #region Incremental test sources.
+        /// <summary>
+        /// Keep in sync with <see cref="BindCallSampleCode"/>.
+        /// </summary>
+        private const string BindCallSampleCodeVariant_ReorderedInvocations = """
```

Comment:
Nit: unless these are being reused, why not just inlined in the tests for easier reading?

Reply:
In general, they are being reused. I'll inline any single-use ones.